controllers/feeds/getOpenGraph.js

```diff
@@ -1,21 +1,34 @@
 const {Request, Response} = require('express');
 const {DomainPage} = require('../../databases/models');
-const OpenGraph = require('open-graph');
+const ogs = require('open-graph-scraper');
 
-const __getOpenGraphInfo = (url) => {
-  return new Promise((resolve, reject) => {
-    OpenGraph(url, (err, meta) => {
-      if (err)
-        return reject({
-          success: false,
+const __getOpenGraphInfo = async (url) => {
+  const options = {url: url};
+  try {
+    return await ogs(options)
+      .then((data) => {
+        const {error, result} = data;
+        if (error) {
+          return {
+            success: false,
+            error: error
+          };
+        }
+
+        return {
+          success: true,
+          data: result
+        };
+      })
+      .catch((err) => {
+        return {
+          success: true,
           error: err
-        });
-      return resolve({
-        success: true,
-        data: meta
+        };
       });
-    });
-  });
+  } catch (e) {
+    console.log(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `__getOpenGraphInfo` function has been updated to use the `open-graph-scraper` package. However, there is a redundancy in error handling. The `await ogs(options)` is already wrapped inside a try-catch block, so there's no need to chain `.then().catch()` on it. This can be simplified by directly using the result from `await ogs(options)`. Also, the catch block at line 23 returns `success: true` when an error occurs, which is misleading.

```diff
-   try {
-     return await ogs(options)
-       .then((data) => {
-         const {error, result} = data;
-         if (error) {
-           return {
-             success: false,
-             error: error
-           };
-         }
- 
-         return {
-           success: true,
-           data: result
-         };
-       })
-       .catch((err) => {
-         return {
-           success: true,
-           error: err
-         };
-        });
-   } catch (e) {
-     console.log(e);
-   }
+   try {
+     const {error, result} = await ogs(options);
+     if (error) {
+       return {
+         success: false,
+         error: error
+       };
+     }
+
+     return {
+       success: true,
+       data: result
+     };
+   } catch (e) {
+     return {
+       success: false,
+       error: e
+     };
+   }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->